tasks/homework.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+)
+
+type RodentType string
+
+const (
+	Mice       RodentType = "Mice"
+	Rat        RodentType = "Rats"
+	Squirrel   RodentType = "Squirrels"
+	Beaver     RodentType = "Beaver"
+	Vole       RodentType = "Vole"
+	GuineaPigs RodentType = "GuineaPigs"
+	Jerboas    RodentType = "Jerboas"
+	Porcupine  RodentType = "Porcupine"
+)
+
+type Location struct {
+	SectorID       SectorID
+	IsSensorPassed bool
+	Times          DayTime
+}
+
+type Rodent struct {
+	ID        int
+	Type      RodentType
+	Morning   *Location
+	Afternoon *Location
+	Evening   *Location
+}
+
+func (r *Rodent) isMovingOnTime(dayTime DayTime) bool {
+	switch dayTime {
+	case Morning:
+		return r.Morning != nil
+	case Afternoon:
+		return r.Afternoon != nil
+	case Evening:
+		return r.Evening != nil
+	}
+
+	return false
+}
+
+func (r *Rodent) moveToSector(sector Sector, dayTime DayTime) {
+	switch dayTime {
+	case Morning:
+		r.Morning = &Location{sector.SectorID, sector.HasSensor, dayTime}
+	case Afternoon:
+		r.Afternoon = &Location{sector.SectorID, sector.HasSensor, dayTime}
+	case Evening:
+		r.Evening = &Location{sector.SectorID, sector.HasSensor, dayTime}
+	}
+}
+
+func (r *Rodent) hasPassedThroughTheSensor() bool {
+	return (r.Morning != nil && r.Morning.IsSensorPassed) ||
+		(r.Afternoon != nil && r.Afternoon.IsSensorPassed) ||
+		(r.Evening != nil && r.Evening.IsSensorPassed)
+}
+
+type DayTime string
+
+const (
+	Morning   DayTime = "Morning"
+	Afternoon DayTime = "Afternoon"
+	Evening   DayTime = "Evening"
+)
+
+type SectorID string
+
+const (
+	TopLeft     SectorID = "TopLeft"
+	TopRight    SectorID = "TopRight"
+	BottomLeft  SectorID = "BottomLeft"
+	BottomRight SectorID = "BottomRight"
+	Center      SectorID = "Center"
+)
+
+type SectorState struct {
+	Times   DayTime
+	Rodents []*Rodent
+}
+
+func (s *SectorState) addRodent(rodent *Rodent) {
+	s.Rodents = append(s.Rodents, rodent)
+}
+
+type Sector struct {
+	SectorID  SectorID
+	HasSensor bool
+	Morning   SectorState
+	Afternoon SectorState
+	Evening   SectorState
+}
+
+func (s *Sector) addState(state SectorState) {
+	switch state.Times {
+	case Morning:
+		s.Morning = state
+	case Afternoon:
+		s.Afternoon = state
+	case Evening:
+		s.Evening = state
+	}
+}
+
+func main() {
+	rodents := []Rodent{
+		{ID: 1, Type: Mice},
+		{ID: 2, Type: Rat},
+		{ID: 3, Type: Squirrel},
+		{ID: 4, Type: Beaver},
+		{ID: 5, Type: Vole},
+		{ID: 6, Type: GuineaPigs},
+		{ID: 7, Type: Jerboas},
+		{ID: 8, Type: Porcupine},
+	}
+
+	sectors := []Sector{
+		{SectorID: TopLeft, HasSensor: false, Morning: SectorState{Times: Morning}, Afternoon: SectorState{Times: Afternoon}, Evening: SectorState{Times: Evening}},
```

Comment:
Важкувато читати та модифікувати. Пропоную ось так:
```suggestion
		{
			SectorID:  TopLeft,
			HasSensor: false,
			Morning:   SectorState{Times: Morning},
			Afternoon: SectorState{Times: Afternoon},
			Evening:   SectorState{Times: Evening},
		},
```